[docs] clarify load return value shape (#6313)
sveltejs/kit: documentation/docs/05-load.md

A [`+page.svelte`](/docs/routing#page-page-svelte) or [`+layout.svelte`](/docs/routing#layout-layout-svelte) gets its `data` from a `load` function.

> If the `load` function is defined in `+page.js` or `+layout.js` it will run both on the server and in the browser. If it's instead defined in `+page.server.js` or `+layout.server.js` it will only run on the server, in which case it can (for example) make database calls and access private [environment variables](/docs/modules#$env-static-private), but can only return data that can be serialized as JSON. On both the client and the server the output needs to be a dictionary at the top level.

Reviewer: I'm not sure how universally understood the term "dictionary" is. It's a term I've more heard in the context of other programming languages. Maybe "key-value object" or something like that? I do think it would be good to mention that this is because we merge the data at different levels, and so this only makes sense if it's a regular object.

### Output

> On the client, the only requirement for the returned `data` is to be an object at the top level. On the server, the output also needs to be JSON-serializable. The exception to this rule are top-level promises, which will be resolved before returning. This makes it easy to return multiple promises without creating a waterfall:

Reviewer 2: I think we need to avoid 'client' terminology, since `+page.js#load` will run on both sides:

```suggestion
The returned `data`, if any, must be an object of values. For a server-only `load` function, these values must be JSON-serializable. Top-level promises will be awaited, which makes it easy to return multiple promises without creating a waterfall:
```

Author: Third time's the charm, this one's great 👍

> On the client, the only requirement for the return value is to be a dictionary at the top level. On the server, the output also needs to be JSON-serializable. On both the client and server any promises on the returned `data` object will be resolved, if they are top-level properties. This makes it easy to return multiple promises without creating a waterfall:

Reviewer 2: I think the output stuff should go _after_ the promise discussion, because it's weird to say 'it has to be JSON-serializable' immediately followed by 'the top-level properties can be promises'. I think we could change this sentence back, and specify the constraints after the code sample

```suggestion
Any promises on the returned `data` object will be resolved, if they are top-level properties. This makes it easy to return multiple promises without creating a waterfall:
```

Author: Yeah that's better.
